core/creators/themedui.js

```diff
@@ -384,6 +388,71 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayDetached` to be `true`.
+	//
+	// If `config.registerCallback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayDetached: true,
+	//		delayDetachedFrequency: 50,
+	//		registerCallback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	// @param {Number} mode CKEditor mode.
+	function delayCreationOnDetachedElement( element, config, data, mode ) {
+		if ( !config || !config.delayDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		var callback = CKEDITOR.config.registerCallback;
+
+		var payload = function() {
+			var delay = parseFloat( config.delay );
+			delay = isNaN( delay ) ? CKEDITOR.config.delayDetachedFrequency : delay;
+
+			var intervalId = setInterval( function() {
+				if ( !element.isDetached() ) {
+					clearInterval( intervalId );
+
+					createInstance( element, config, data, mode );
+				}
+			}, delay );
+		};
+
+		var registerPayload = function() {
+			if ( !element.isDetached() ) {
+				createInstance( element, config, data, mode );
+			} else {
+				// Element still not ready!
+			}
+		};
+
+		if ( config.registerCallback ) {
+			if ( typeof config.registerCallback !== 'function' ) {
+				CKEDITOR.error( 'invalid-callback', { callback: config.registerCallback } );
+			} else {
+				payload = registerPayload;
+				callback = config.registerCallback;
+			}
+		}
```

Comment:
This logic can be simplified if you implement an early return proposed in the other review comment (sorry but I can't link it during review).